hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/ratis/RatisHelper.java

```diff
@@ -170,8 +173,12 @@ public static RaftGroup newRaftGroup(Pipeline pipeline) {
   public static RaftClient newRaftClient(RpcType rpcType, Pipeline pipeline,
       RetryPolicy retryPolicy, GrpcTlsConfig tlsConfig,
       ConfigurationSource ozoneConfiguration) throws IOException {
+    // TODO: For now, always set first node as primary. In the future we should
+    //  use the local or closest node in the pipline.
+    final DatanodeDetails leader = pipeline.getLeaderNode();
```

Comment:
The TODO can be removed since the first datanode is supposed to be the closest datanode.  SCM is supposed to sort the datanode in the pipeline according the distance to the client.  Not sure if SCM is currently sorting the datanodes.  We should check the code.

BTW, the leader local variable is no longer needed.  We should revert the change.